rust/processor/src/processors/default_processor.rs

```diff
@@ -76,76 +82,84 @@ async fn insert_to_db(
         insert_block_metadata_transactions_query,
         block_metadata_transactions,
         get_config_table_chunk_size::<BlockMetadataTransactionModel>(
             "block_metadata_transactions",
             per_table_chunk_sizes,
         ),
-    );
-    let wst_res = execute_in_chunks(
-        conn.clone(),
-        insert_write_set_changes_query,
-        wscs,
-        get_config_table_chunk_size::<WriteSetChangeModel>(
-            "write_set_changes",
-            per_table_chunk_sizes,
-        ),
-    );
-    let mm_res = execute_in_chunks(
+    )
+    .await?;
+
+    execute_in_chunks(
         conn.clone(),
         insert_move_modules_query,
         move_modules,
         get_config_table_chunk_size::<MoveModule>("move_modules", per_table_chunk_sizes),
-    );
-
-    let mr_res = execute_in_chunks(
-        conn.clone(),
-        insert_move_resources_query,
-        move_resources,
-        get_config_table_chunk_size::<MoveResource>("move_resources", per_table_chunk_sizes),
-    );
+    )
+    .await?;
 
-    let ti_res = execute_in_chunks(
+    execute_in_chunks(
         conn.clone(),
         insert_table_items_query,
         table_items,
         get_config_table_chunk_size::<TableItem>("table_items", per_table_chunk_sizes),
-    );
+    )
+    .await?;
 
-    let cti_res = execute_in_chunks(
+    execute_in_chunks(
         conn.clone(),
         insert_current_table_items_query,
         current_table_items,
         get_config_table_chunk_size::<CurrentTableItem>(
             "current_table_items",
             per_table_chunk_sizes,
         ),
-    );
+    )
+    .await?;
 
-    let tm_res = execute_in_chunks(
+    execute_in_chunks(
         conn.clone(),
         insert_table_metadata_query,
         table_metadata,
         get_config_table_chunk_size::<TableMetadata>("table_metadatas", per_table_chunk_sizes),
-    );
+    )
+    .await?;
 
-    let (txns_res, bmt_res, wst_res, mm_res, mr_res, ti_res, cti_res, tm_res) =
-        join!(txns_res, bmt_res, wst_res, mm_res, mr_res, ti_res, cti_res, tm_res);
+    if !deprecated_tables.contains(&"transactions".to_string()) {
+        execute_in_chunks(
+            conn.clone(),
+            insert_transactions_query,
+            txns,
+            get_config_table_chunk_size::<TransactionModel>("transactions", per_table_chunk_sizes),
+        )
+        .await?;
+    };
+
+    if !deprecated_tables.contains(&"write_set_changes".to_string()) {
+        execute_in_chunks(
+            conn.clone(),
+            insert_write_set_changes_query,
+            wscs,
+            get_config_table_chunk_size::<WriteSetChangeModel>(
+                "write_set_changes",
+                per_table_chunk_sizes,
+            ),
+        )
+        .await?;
+    };
 
-    for res in [
-        txns_res, bmt_res, wst_res, mm_res, mr_res, ti_res, cti_res, tm_res,
-    ] {
-        res?;
-    }
+    if !deprecated_tables.contains(&"move_resources".to_string()) {
```

Comment:
this is expensive- instead of strings, lets use an enum for this type and https://github.com/bitflags/bitflags, instead of array iteration/search for each txn*structs

Reply:
it's a hashset lookup. Should be O(1). The to_string is unnecessary though.